mono/metadata/icall.c

```diff
@@ -3582,167 +3582,174 @@ ves_icall_RuntimeMethodInfo_GetGenericArguments (MonoReflectionMethodHandle ref_
 	return res;
 }
 
-MonoObject *
-ves_icall_InternalInvoke (MonoReflectionMethod *method, MonoObject *this_arg, MonoArray *params, MonoException **exc) 
+MonoObjectHandle
+ves_icall_InternalInvoke (MonoReflectionMethodHandle method_handle, MonoObjectHandle this_arg_handle,
+			  MonoArrayHandle params_handle, MonoExceptionHandleOut exception_out, MonoError* error)
 {
-	ERROR_DECL (error);
+	MonoReflectionMethod* const method = MONO_HANDLE_RAW (method_handle);
+	MonoObject* const this_arg = MONO_HANDLE_RAW (this_arg_handle);
+	MonoArray* const params = MONO_HANDLE_RAW (params_handle);
+
 	/* 
 	 * Invoke from reflection is supposed to always be a virtual call (the API
 	 * is stupid), mono_runtime_invoke_*() calls the provided method, allowing
 	 * greater flexibility.
 	 */
-	MonoMethod *m = method->method;
-	MonoMethodSignature *sig = mono_method_signature_internal (m);
-	MonoImage *image;
-	int pcount;
-	void *obj = this_arg;
-
-	*exc = NULL;
+	MonoMethod* m = method->method;
+	MonoMethodSignature* const sig = mono_method_signature_internal (m);
+	MonoImage* image = NULL;
+	int pcount = 0;
+	void* obj = this_arg;
+	char* this_name = NULL;
+	char* target_name = NULL;
+	char* msg = NULL;
+	MonoObject* result = NULL;
+	MonoArray* arr = NULL;
+	MonoException* exception = NULL;
+
+	*MONO_HANDLE_REF (exception_out) = NULL;
 
 	if (mono_security_core_clr_enabled () &&
 	    !mono_security_core_clr_ensure_reflection_access_method (m, error)) {
-		mono_error_set_pending_exception (error);
-		return NULL;
+		goto return_null;
 	}
```

Comment:
Whats the point of this change ?

Reply:
Single point of return is often useful, but less critical in this case. I can probably reduce the change.

Reply:
I've left it alone for now.

Reply:
By "left it alone" I mean, "I left my change in".